lib/shared.bash

```diff
@@ -164,10 +164,29 @@ function build_image_override_file_with_version() {
     service_name=$1
     image_name=$2
     target=$3
-    cache_from_amt=$4
-    shift 4
+    shift 3
+
+    # load cache_from array
+    cache_from=()
+    if (( $# > 0 )); then
+      cache_from_amt=$1
+      shift
+      while (( cache_from_amt-- > 0 )); do
+        cache_from+=( "$1" ); shift
+      done
+    fi
+
+    # load labels array
+    labels=()
+    if (( $# > 0 )); then
+      labels_amt=$1
+      shift
+      while (( labels_amt-- > 0 )); do
+        labels+=( "$1" ); shift
+      done
+    fi
 
-    if [[ -z "$image_name" ]] && [[ -z "$target" ]] && [[ "$cache_from_amt" -eq 0 ]]; then
+    if [[ -z "$image_name" ]] && [[ -z "$target" ]] && [[ "${#cache_from[@]}" -eq 0 ]] && [[ "${#labels[@]}" -eq 0 ]]; then
```

Comment:
With the previous changes I suggested, the `_amt` variables would always exist so this could be:

```suggestion
    if [[ -z "$image_name" ]] && [[ -z "$target" ]] && [[ "$cache_from_amt" -eq 0 ]] && [[ "$labels_amt" -eq 0 ]]; then
```